internal/transformer/classes/generators/dockerfilegenerators/golangdockerfilegenerator.go

```diff
@@ -0,0 +1,165 @@
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+package dockerfilegenerators
+
+import (
+	"io/ioutil"
+	"path/filepath"
+
+	"github.com/konveyor/move2kube/environment"
+	"github.com/konveyor/move2kube/internal/common"
+
+	"github.com/konveyor/move2kube/types/qaengine/commonqa"
+	transformertypes "github.com/konveyor/move2kube/types/transformer"
+	"github.com/konveyor/move2kube/types/transformer/artifacts"
+	"github.com/sirupsen/logrus"
+	"golang.org/x/mod/modfile"
+	"golang.org/x/mod/module"
+)
+
+const (
+	goVersion = "1.16"
+)
+
+// GolangDockerfileGenerator implements the Transformer interface
+type GolangDockerfileGenerator struct {
+	Config transformertypes.Transformer
+	Env    *environment.Environment
+}
+
+// Init Initializes the transformer
+func (t *GolangDockerfileGenerator) Init(tc transformertypes.Transformer, env *environment.Environment) (err error) {
+	t.Config = tc
+	t.Env = env
+	return nil
+}
+
+// GetConfig returns the transformer config
+func (t *GolangDockerfileGenerator) GetConfig() (transformertypes.Transformer, *environment.Environment) {
+	return t.Config, t.Env
+}
+
+// BaseDirectoryDetect runs detect in base directory
+func (t *GolangDockerfileGenerator) BaseDirectoryDetect(dir string) (namedServices map[string]transformertypes.ServicePlan, unnamedServices []transformertypes.TransformerPlan, err error) {
+	return nil, nil, nil
+}
+
+// DirectoryDetect runs detect in each sub directory
+func (t *GolangDockerfileGenerator) DirectoryDetect(dir string) (namedServices map[string]transformertypes.ServicePlan, unnamedServices []transformertypes.TransformerPlan, err error) {
+	modFilePath := filepath.Join(dir, "go.mod")
+	data, err := ioutil.ReadFile(modFilePath)
+	if err != nil {
+		logrus.Errorf("Error while reading the go.mod file : %s", err)
+		return nil, nil, nil
+	}
+	modFile, err := modfile.Parse(modFilePath, data, nil)
+	if err != nil {
+		logrus.Errorf("Error while parsing the go.mod file : %s", err)
+		return nil, nil, nil
+	}
+	if modFile.Go == nil {
+		logrus.Debugf("Didn't find the Go version in the go.mod file at path %s, selecting Go version %s", modFilePath, goVersion)
+		modFile.Go.Version = goVersion
+	}
+	prefix, _, ok := module.SplitPathVersion(modFile.Module.Mod.Path)
+	if !ok {
+		logrus.Errorf("Invalid module path")
+		return nil, nil, nil
+	}
+	serviceName := filepath.Base(prefix)
+	namedServices = map[string]transformertypes.ServicePlan{
+		serviceName: []transformertypes.TransformerPlan{{
+			Mode:              t.Config.Spec.Mode,
+			ArtifactTypes:     []transformertypes.ArtifactType{artifacts.ContainerBuildArtifactType},
+			BaseArtifactTypes: []transformertypes.ArtifactType{artifacts.ContainerBuildArtifactType},
+			Paths: map[string][]string{
+				artifacts.ProjectPathPathType: {dir},
+			},
+			Configs: map[string]interface{}{
+				artifacts.DockerfileTemplateConfigConfigType: map[string]interface{}{
```

Comment:
Would be good to define a fixed new config struct instead of map[string]interface{} if we are anyway going with a custom transform function.